File: packages/train/src/game/randomBoard.ts
import { array2d, randomShuffle, range } from '../utils';
import { Board, DECK, EMPTY } from './types';

export const randomBoard = (): Board => {
    const deck = [...DECK];
    const numColumns = 8;
    const shuffled = randomShuffle(deck);
    const layout = array2d([23, numColumns], EMPTY);

    let level = 1; // level === 0 is place & bin
    while (shuffled.length > 0) {
        for (const column of range(numColumns)) {
            layout[level][column] = shuffled.pop() || EMPTY;
        }

        level += 1;
    }

    return layout;
};

File: pythonFramework/game/random_board.py
import numpy as np
from copy import deepcopy

from .constants import DECK, EMPTY


def randomBoard():
    deck = deepcopy(DECK)
    np.random.shuffle(deck)
    numColumns = 8
    layout = np.full([23, numColumns], EMPTY).tolist()

    level = 1  # level === 0 is place & bin
    while len(deck) > 0:
        for column in range(numColumns):
            try:
                layout[level][column] = deck.pop()
            except:
                pass
            finally:
                pass

        level += 1

    return layout
